certbot-dns-google/certbot_dns_google/dns_google.py

```diff
@@ -216,23 +220,32 @@ def get_existing_txt_rrset(self, zone_id, record_name):
         """
         Get existing TXT records from the RRset for the record name.
 
+        If an error occurs while requesting the record set, it is suppressed
+        and None is returned.
+
         :param str zone_id: The ID of the managed zone.
         :param str record_name: The record name (typically beginning with '_acme-challenge.').
 
-        :returns: List of TXT record values
-        :rtype: `list` of `string`
+        :returns: List of TXT record values or None
+        :rtype: `list` of `string` or `None`
 
         """
         rrs_request = self.dns.resourceRecordSets()  # pylint: disable=no-member
         request = rrs_request.list(managedZone=zone_id, project=self.project_id)
-        response = request.execute()
         # Add dot as the API returns absolute domains
         record_name += "."
-        if response:
-            for rr in response["rrsets"]:
-                if rr["name"] == record_name and rr["type"] == "TXT":
-                    return rr["rrdatas"]
-        return []
+        try:
+            response = request.execute()
+        except googleapiclient_errors.Error:
+            logger.info("Unable to list existing records. If you're "
+                        "requesting a wildcard certificate, this might not work.")
+            logger.debug("Error was:", exc_info=True)
```

Comment:
This error gets a bit "lost" in the very verbose output of the library we're using. It's a noop when not requesting a certificate with wildcard + base domain, so `logger.warn` is not suitable for use here. However if the user does request a wildcard + base domain certificate, what would you think about raising a flag here, and if the flag is present when failing, including this message on a `logger.warn` level with the actual API error?